src/lib/circuit_value.ts

```diff
@@ -239,3 +239,26 @@ export function circuitMain(
     Array.from(publicIndexSet).map((i) => paramTypes[i])
   );
 }
+
+export function toFieldsMagic(thing: any): Field[] {
+  if (typeof thing == 'object') {
+    if (thing.toFields) {
+      // console.log('toFields');
+      return thing.toFields();
+    }
+
+    if (Array.isArray(thing)) {
+      // console.log('array', thing.length);
+      return thing.map((x) => toFieldsMagic(x)).flat();
+    }
+
+    let fields = [] as Field[];
+    let keys = Object.keys(thing).sort();
+    for (let key of keys) {
+      // console.log({ key });
+      fields.push(...toFieldsMagic(thing[key]));
+    }
+    return fields;
+  }
+  return [];
```

Comment:
This isn't used anymore, but being able to return `[]` is important to handle mixed objects, which contain both circuit- and non-circuit properties. For example, a `party.body.update.verificationKey` contains a property `value: string`, which holds the base58-formatted verification key. This string should be ignored by `toFieldsMagic()`, so that a Party can be turned into field elements.

Reply:
Accumulating the skipped data could be an option!

Reply:
I decided to remove the function instead of changing it (https://github.com/o1-labs/snarkyjs/pull/123/commits/b80d39ffb3b32e69716bbce5a0f5428e408c131f), since it's unused and most likely won't be useful, because if we want to compute hash inputs we'll have to conform to the more elaborate system used in Mina.